src/login/logind.c

```diff
@@ -1090,6 +1090,10 @@ static int manager_startup(Manager *m) {
         if (r < 0)
                 log_warning_errno(r, "Button enumeration failed: %m");
 
+        r = manager_load_scheduled_shutdown(m);
+        if (r < 0)
+                log_warning_errno(r, "Load scheduled shutdown failed: %m");
```

Comment:
for new code we try to be strict in the rules who logs. i.e. for a function figure out whether it shall log about all errors, or about none (usually "library"-like functions should not log on their own and leave that to the caller, while "main-program"-like functions should log on their own, but the line between the two is blurred). Then stick to it, and then either let the caller log about it (in case the function is non-logging) or do not log in the caller (in case it logs on its own).

manager_load_scheduled_shutdown() logs about most errors already, and it's probably more "main-program-like", hence it should log about all. Because of that this log line here in the invocation should just go away.

Following this rule ensures we log about everything there is to log about exactly once — and not twice or not at all.

We didn't always follow this rule, so you'll find place where we are sloppy on this (in particular in logind there are plenty), but for new code please always follow this rule.

See: https://systemd.io/CODING_STYLE/#logging